11-Backbonejs/Backbone-Views.md

```diff
@@ -0,0 +1,321 @@
+  }
+```
+
+Again this looks very much like what we originally wrote in `app.js`  We did change the code to use `this.$` instead of direct jQuery and called `readNewTaskForm()` as an instance method with `this.`.  
+
+**Note you will need to delete the original event handlers in `app.js`.**
+
+**Gah!** The Add button ALSO doesn't work anymore!
+
+Good point, the delete and add buttons are actually working, and tasks are getting added and removed from the collection, but the problem is we're not re-rending the list.
+
+## Adding a Backbone Event Listener
+
+If we add an event listener to the collection we can call render when the collection is updated.  
+
+Similar to `on` we can use a method called `listenTo` to add an event listener for our collection.  We will do so in the `initialize` method.
+
+```javascript
+  initialize: function(params) {
+    this.template = params.template;
+
+    this.listenTo(this.model, "update", this.render);
+  },
+```
+
+## Last bit, adjusting styles
+
+You may have noticed that the styling is a bit broken.  That's because the `li` tag we set in the `TaskView`  doesn't have any classes added to it for styling.  
+
+**NOTE** this is for `task_view.js` not `task_list_view.js`.  
+
+
+```javascript
+// src/views/task_view.js
+
+  initialize: function(params) {
+    this.template = params.template;
+    
+    this.listenTo(this.model, "update", this.render);
+    
+	 // Add classes for styling
+    this.$el.addClass("task-item");
+    this.$el.addClass("column");
+    this.$el.addClass("column-block");
+  },
+```
+
+We also need to make one adjustment by removing the `li` tag from the HTML template.
+
+## Summary
+
+So now we have a solution which displays our list of tasks, lets us toggle them complete or incomplete, delete and create tasks.  The view acts as a coordinator handling events and coordinating the response with the template & the model.  
+
+
+## Optimizations & Extensions
+
+What could we do to improve on this?  There are a few things we could change.  Noice we recreate all the `TaskView`s every time the `TaskListView` is rendered.  We could store them in an array to avoid recreating them.  We could also create some way to expand a task, viewing it's description when clicked on.  
+
+## Last Note on Coding Style
+
+Also notice that we only used `$el` to select items inside a view.  Views should never use jQuery directly `$()`, but rather use `$el` to ensure that we only select items within the view.  Some developers make an exception for selecting templates.  
+
+## What Did We Accomplish?
+- Create a basic Backbone view to display a task. It had one function:
+  - `initialize()` is run once to set everything up
+  - `render()` generates HTML, and may be run many times
+- Use the underscore templating engine to separate concerns and clean up our rendering code.
+- Added event handler methods to respond to click events.
+- Create a more complex Backbone view to manage our whole collection
```

Comment:
Thoughts on a separate resource which will highlight all of the pieces that we've built, where they live within the overall application and how they're all interacting with one another?